### ISSUE TITLE
Add an option to delete forward result in softmax cross entropy

### ISSUE THREAD

**Issue by unnonouno** [pfnet/chainer#1067]
When the number of output labels is large especially in NLP, forward result of softmax cross entropy `y` consumes a lot of memory. I want an option to delete this value after forward computation.
